### PR TITLE
[CXF-9006] Skip WSS40 cer validation check when on Red Hat JDK

### PR DESCRIPTION
**Pull request by jgoodyear**
This change will allow the WSS40 cert check test to be skipped when using Red Hat JDK.

Red Hat OpenJDK must have RSA keySize 2048 or greater.
see jdk.certpath.disabledAlgorithms=MD2, SHA1, MD5, DSA, RSA keySize < 2048

### CODE REVIEW

File: services/xkms/xkms-x509-handlers/src/test/java/org/apache/cxf/xkms/x509/validator/TrustedAuthorityValidatorCRLTest.java [hunk 1]
```diff
@@ -91,8 +91,12 @@ public void testIsCertChainValid() throws CertificateException {
                           validator.isCertificateChainValid(Arrays.asList(certificateRoot)));
         Assert.assertTrue("wss40rev should not be valid",
                           !validator.isCertificateChainValid(Arrays.asList(certificateWss40Rev)));
-        Assert.assertTrue("wss40 should be valid",
-                          validator.isCertificateChainValid(Arrays.asList(certificateWss40)));
+        //Red Hat OpenJDK must have RSA keySize 2048 or greater.
+        //see jdk.certpath.disabledAlgorithms=MD2, SHA1, MD5, DSA, RSA keySize < 2048
```
Comment (reta) [inline services/xkms/xkms-x509-handlers/src/test/java/org/apache/cxf/xkms/x509/validator/TrustedAuthorityValidatorCRLTest.java:95]: @jgoodyear I am wondering if this should be really treated as RedHat specific vs checking the `AlgorithmConstraints`:

```
final AlgorithmConstraints disabledAlgConstraints =
            new DisabledAlgorithmConstraints(PROPERTY_CERTPATH_DISABLED_ALGS);

```

Comment (jgoodyear) [reply services/xkms/xkms-x509-handlers/src/test/java/org/apache/cxf/xkms/x509/validator/TrustedAuthorityValidatorCRLTest.java:95]: Given we can not change JVM Parameters at runtime for the test case we have 2 options:
1. Skip on RH JVM,
2. Generate a RSA >= 2048 certificate to replace the wss40 cer we currently have.

I've tried some self signed certs, however i keep getting invalid cert errors - may need to reach out to Colm.

Comment (ffang) [reply services/xkms/xkms-x509-handlers/src/test/java/org/apache/cxf/xkms/x509/validator/TrustedAuthorityValidatorCRLTest.java:95]: Hi @jgoodyear ,

Thanks for reporting and the investigation!

Could you please try out the new cer files I put on
https://issues.apache.org/jira/browse/CXF-9006
which are RSA2048 instead previous DSA 1024.

Please let me know if they are helpful.

Best Regards
Freeman

Comment (jgoodyear) [reply services/xkms/xkms-x509-handlers/src/test/java/org/apache/cxf/xkms/x509/validator/TrustedAuthorityValidatorCRLTest.java]: The updated certificates allowed the test case to pass on Stream 9 OS on POWER.
I'm running some builds on other platforms/JVMs to ensure all is well.

Comment (jgoodyear) [reply services/xkms/xkms-x509-handlers/src/test/java/org/apache/cxf/xkms/x509/validator/TrustedAuthorityValidatorCRLTest.java]: Test suite testing:

Windows 11 Pro, x64 , IBM Semeru Java 17: Passed
Ubuntu 22.04 LTS, x64, Eclipse Adoptium  Java 17: Passed
MacOS 14.4.1, AArch64, IBM Semeru Java 17: Passed
Stream 9 OS, PPC64LE, Red Hat OpenJDK 17: Passed

Comment (ffang) [reply services/xkms/xkms-x509-handlers/src/test/java/org/apache/cxf/xkms/x509/validator/TrustedAuthorityValidatorCRLTest.java]: Awesome, thanks for the update! I will commit new cer files.

**Review by ffang — APPROVED** [review]

**Pull request merged**